db/snapshot.go

```diff
@@ -18,9 +18,18 @@ func (c *Collection) GetSnapshot() (*Snapshot, error) {
 	if err != nil {
 		return nil, err
 	}
+	c.indexMut.RLock()
+	indexes := make([]*Index, len(c.indexes))
+	copy(indexes, c.indexes)
+	defer c.indexMut.RUnlock()
```

Comment:
Nit: Shouldn't this simply unlock here since we're done copying the indexes?

Reply:
Nice catch! Yes that was my intention.